src/lib/spinel/max_power_table.hpp

```diff
@@ -26,53 +26,86 @@
  *  POSSIBILITY OF SUCH DAMAGE.
  */
 
-#ifndef OT_POSIX_PLATFORM_MAX_POWER_TABLE_HPP_
-#define OT_POSIX_PLATFORM_MAX_POWER_TABLE_HPP_
+#ifndef OT_LIB_SPINEL_MAX_POWER_TABLE_HPP_
+#define OT_LIB_SPINEL_MAX_POWER_TABLE_HPP_
 
 #include "core/radio/radio.hpp"
 
 namespace ot {
-namespace Posix {
+namespace Spinel {
```

Comment:
This is moving this file under `spinel` lib and `Spinel` namespace.
The `MaxPowerTable` seems to be fully unrelated to spinel itself and more of radio driver thing.

Can we consider putting this in another lib/location? (maybe same place as before under posix?)